src/components/EnergiaTodistus/ET2018Form.svelte

```diff
@@ -42,14 +43,69 @@
     api.laatimisvaiheet
   );
 
-  let isValidForm = false;
-  $: isValidForm = et.isValidForm(schema, energiatodistus);
-  $: console.log('Form validation: ', isValidForm);
+  let kayttotarkoitusluokat = Either.Left('Not initialized');
+  Future.fork(
+      _ => {},
+      result => kayttotarkoitusluokat = Either.Right(result),
+      api.kayttotarkoitusluokat2018
+  );
+
+  let alakayttotarkoitusluokat = Either.Left('Not initialized');
+  Future.fork(
+      _ => {},
+      result => alakayttotarkoitusluokat = Either.Right(result),
+      api.alakayttotarkoitusluokat2018
+  );
+
+  let kayttotarkoitusluokkaId = Maybe.None();
+  $: if (kayttotarkoitusluokkaId.isSome() &&
+      !kayttotarkoitusluokkaId.equals(et.findKayttotarkoitusluokkaId(
+        energiatodistus.perustiedot.kayttotarkoitus, alakayttotarkoitusluokat))) {
+
+    energiatodistus = R.set(R.lensPath(['perustiedot', 'kayttotarkoitus']),
+        R.compose(
+          Either.orSome(Maybe.None()),
+          Either.map(alaluokat => R.length(alaluokat) === 1 ?
+              Maybe.Some(alaluokat[0].id) : Maybe.None()),
```

Comment:
Olis mahdollinen tehdä tämä koko lambda näin.
`R.compose(
  R.map(R.prop('id')),
  Maybe.fromNull,
  R.nth(1)
)`

Reply:
ei koska valinta voidaan tehdä automaattisesti vain jos valittavia alaluokkia on yksi

Reply:
tarkoitus ei ole aina valita headia paitsi siinä tapauksia että listassa on vain yksi itemi

Reply:
jos alaluokkia on useita niin silloin on tarkoitus jättää valinta tyhjäksi